app/src/shared/components/TextField/NumberField/index.jsx

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import React, { useCallback, useRef } from 'react'
+import cx from 'classnames'
+
+import SvgIcon from '$shared/components/SvgIcon'
+
+import styles from '../textField.pcss'
+
+type Props = {
+    className?: string,
+    onAutoComplete?: (boolean) => void,
+    type?: string,
+    min?: number,
+    max?: number,
+    step?: number,
+    hideButtons?: boolean,
+}
+
+const NumberField = ({ className, onAutoComplete, hideButtons, ...props }: Props) => {
+    const inputRef = useRef(null)
+
+    const onAnimationStart = useCallback(({ animationName }: SyntheticAnimationEvent<EventTarget>) => {
+        if (onAutoComplete && (animationName === styles.onAutoFillStart || animationName === styles.onAutoFillCancel)) {
+            onAutoComplete(animationName === styles.onAutoFillStart)
+        }
+    }, [onAutoComplete])
+
+    const addValue = useCallback((step) => {
+        if (inputRef.current) {
+            const { value } = inputRef.current
+            let valueInt = parseInt(value, 10)
+            let stepInt = parseInt(step, 10)
```

Comment:
This is an integer field, correct? I wonder if we should plan ahead and make it work with floats as well. Thoughts?